cli/azd/pkg/tools/azcli/mock_azcli.go

```diff
@@ -0,0 +1,211 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcli
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/containerregistry/armcontainerregistry"
+)
+
+var _ AzCli = &Mock{}
+
+// Mock can be used to build a mock AzCli implementation based on provided functions
+type Mock struct {
```

Comment:
Curious... Is this approach not taking advantage of `golang`'s duck typing? What if we simply defined the minimal interface that only contained what `azure_resource_manager` depends on `azcli`?

Reply:
Matt and I chatted offline -- we should create the smaller interfaces naturally as part of the `azcli` refactor down the road.